packages/runtime/test/fixtures/app-update/assets/approval_program.py

```diff
@@ -0,0 +1,54 @@
+from pyteal import *
+
+def app():
+    '''
+    - An app with 2 global variables: `total: int` and `counter: int`.
+        - (on app creation), `counter = 0, n = 10`
+    - The app update accepts exactly one argument: `n: int`
+    - On update call the app verifies that: `n >= 1`,  increments `app.counter += 1` and:
+        * when `app.counter % 2 == 0` then update `app.total += n`
+        * otherwise, subtract: `app.total -= n`
+    '''
+    total = Bytes("total")
+    counter = Bytes("counter")
+
+    on_creation = Seq([
+        App.globalPut(counter, Int(0)),
+        App.globalPut(total, Int(10)),
+        Return(Int(1))
+    ])
+
+    common_checks = Assert(And(
+        Txn.application_args.length() == Int(1),
+        Btoi(Txn.application_args[0]) >= Int(1)
+    ))
+
+    on_success = Seq([
+        App.globalPut(total, App.globalGet(total) + Btoi(Txn.application_args[0])),
+        App.globalPut(counter, App.globalGet(counter) + Int(1)),
+        common_checks
```

Comment:
I think we should do `common_checks` before `App.globalPut(..)`

Reply:
i think this will get reverted if checks don't pass right ?
but it will be little faster, will update